prometheus/config.go

```diff
@@ -143,3 +153,22 @@ func (c Configuration) NewReporter(
 
 	return reporter, nil
 }
+
+func (c Configuration) resolveListenAddress() (addr string, resolved bool, err error) {
```

Comment:
Technically you don't really need `resolved` since `addr == "" && !resolved` is always false.